haskell-style.md

```diff
@@ -391,3 +412,134 @@ import qualified Data.List.NonEmpty as NE
 
 import qualified Data.Sequence as Seq
 ```
+
+### Import groups
+
+Imports should be divided into groups:
+
+1. Prelude (when explicit)
+1. Un-`qualified` dependency libraries
+1. `qualified` dependency libraries
+1. Un-`qualified` FrontRow libraries
+1. `qualified` FrontRow libraries
+1. Un-`qualified` modules from this library
+1. `qualified` modules from this libraries
+
+Aside from the empty line between groups, there should be no other whitespace.
+
+```haskell
+-- Bad
+-- What's coming from where?
+import ClassyPrelude
+import Network.AWS.S3
+import Network.AWS
+import Data.Text (Text)
+import Json
+import qualified Data.Text as T
+import Control.Lens
+import Unit
+import Data.Conduit
+import TextAssets.S3
+import Data.Set (Set)
+import qualified Data.Set as Set
+import Data.Map (Map)
+import qualified Data.Map as Map
+
+-- Bad
+-- You can't quickly highlight and :sort things
+import ClassyPrelude
+
+import Network.AWS.S3
+import Network.AWS
+import Data.Text (Text)
+import qualified Data.Text as T
+import Control.Lens
+import Data.Conduit
+import Data.Set (Set)
+import qualified Data.Set as Set
+import Data.Map (Map)
+import qualified Data.Map as Map
+
+import TextAssets.S3
+
+import Json (Json)
+import qualified Json as J
+import Unit
+
+-- Good
+import ClassyPrelude
+
+import Control.Lens
+import Data.Conduit
+import Data.Map (Map)
+import Data.Set (Set)
+import Data.Text (Text)
+import Network.AWS
+import Network.AWS.S3
+
+import qualified Data.Map as Map
+import qualified Data.Set as Set
+import qualified Data.Text as T
+
+import TextAssets.S3
+
+import Json (Json)
+import Unit
+
+import qualified Json as J
+```
+
+## Exports
+
+Use sorted, multi-line exports. There are two exceptions to this rule:
+
+1. A single-line export for `Main`, when it only exports `main`:
+
+   ```haskell
+   module Main (main) where
+   ```
+
+1. If the order of exports matters in your desired Haddock output, you may
+   violate sorting to achieve it.
+
+Otherwise:
+
+```haskell
+-- Bad
+module Driver (scienceOptions, socialStudiesOptions, mainWith) where
+
+-- Good
+module Driver
+  ( mainWith
+  , scienceOptions
+  , socialStudiesOptions
+  ) where
+```
+
+## Declaring Extensions
```

Comment:
Yeah, I'm a huge fan of this. Especially since some extensions can clash and also can complicate errors and inference. What I usually see in practice is that orgs identify a subset of extensions to turn on by default ( `OverloadedStrings`, `LambdaCase`) and then all other extensions must be in a pragma.